File: Library.Core.Web/src/app/view-models/publication-house/postPublication-houseView.ts
export class PostPublicationHouseView {
  constructor(
    public id: number = 0,
    public name?: string,
    public adress?: string,
    public books?: BookPostPublicationHouseViewItem[]
  ) { }
}

export class BookPostPublicationHouseViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/PublicationHouseViewModels/PostPublicationHouseView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.PublicationHouse
{
    public class PostPublicationHouseView
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Adress { get; set; }

        public virtual ICollection<BookPostPublicationHouseViewItem> Books { get; set; }

        public PostPublicationHouseView()
        {
            Books = new List<BookPostPublicationHouseViewItem>();
        }
    }

    public class BookPostPublicationHouseViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public BookPostPublicationHouseViewItem()
        {

        }
    }
}
